src/omnicore/rules.cpp

```diff
@@ -317,41 +339,45 @@ bool ActivateFeature(uint16_t featureId, int activationBlock, int transactionBlo
         (activationBlock > (transactionBlock + params.MAX_ACTIVATION_BLOCKS))) {
-        PrintToLog("Feature activation of ID %d refused due to notice checks\n", featureId);
-        return false;
+            PrintToLog("Feature activation of ID %d refused due to notice checks\n", featureId);
+            return false;
     }
 
     // check feature is recognized and activation is successful
+    std::string featureName;
+    bool supported = OMNICORE_VERSION >= minClientVersion;
     switch (featureId) {
         case FEATURE_CLASS_C:
             MutableConsensusParams().NULLDATA_BLOCK = activationBlock;
-            PrintToLog("Feature activation of ID %d succeeded. "
-                       "Class C transaction encoding is going to be enabled at block %d.\n",
-                       featureId, params.NULLDATA_BLOCK);
-            return true;
+            featureName = "Class C transaction encoding";
+        break;
         case FEATURE_METADEX:
             MutableConsensusParams().MSC_METADEX_BLOCK = activationBlock;
-            PrintToLog("Feature activation of ID %d succeeded. "
-                       "The distributed token exchange is going to be enabled at block %d.\n",
-                       featureId, params.MSC_METADEX_BLOCK);
-            return true;
+            featureName = "Distributed Meta Token Exchange";
+        break;
         case FEATURE_BETTING:
             MutableConsensusParams().MSC_BET_BLOCK = activationBlock;
-            PrintToLog("Feature activation of ID %d succeeded. "
-                       "Bet transactions are going to be enabled at block %d.\n",
-                       featureId, params.MSC_BET_BLOCK);
-            return true;
+            featureName = "Bet transactions";
+        break;
         case FEATURE_GRANTEFFECTS:
             MutableConsensusParams().GRANTEFFECTS_FEATURE_BLOCK = activationBlock;
-            PrintToLog("Feature activation of ID %d succeeded. "
-                       "The potential side effect of crowdsale participations, when "
-                       "granting tokens, is going to be disabled at block %d.\n",
-                       featureId, params.GRANTEFFECTS_FEATURE_BLOCK);
-            return true;
+            featureName = "Remove grant side effects";
+        break;
+        default:
+            featureName = "Unknown feature";
+            supported = false;
+        break;
     }
-
-    PrintToLog("Feature activation of id %d refused due to unknown feature ID\n", featureId);
-    return false;
+    PrintToLog("Feature activation of ID %d processed. %s will be enabled at block %d.\n", featureId, featureName, activationBlock);
+    AddPendingActivation(featureId, activationBlock, minClientVersion, featureName);
+    AddAlert("omnicore", ALERT_BLOCK_EXPIRY, activationBlock, strprintf("Feature %d ('%s') will go live at block %d", featureId, featureName, activationBlock));
+    if (!supported) {
```

Comment:
Shoot - you're right - nice find...  Yeah so if the alert hasn't expired the iterator isn't advanced and it'll just loop the same one endlessly - good catch mate thank you :)